tests/setup-permissions.test.sh

```
@@ -0,0 +1,123 @@
+
+# ── 5. Apply ────────────────────────────────────────────────────────────────
+bash "$SCRIPT" --apply --root "$SB" --user "$CUR_USER" --group "$CUR_GROUP" >/dev/null 2>&1
+
+# 5a. writable dirs created
+_missing=0
+for d in storage/logs storage/cache storage/tmp storage/backups public/uploads \
+         public/uploads/copertine public/uploads/archives/covers data/dewey \
+         writable/uploads locale storage/uploads/plugins; do
+    [ -d "$SB/$d" ] || { _missing=1; echo "      missing: $d"; }
+done
+[ "$_missing" -eq 0 ] && ok "all writable data directories created" || bad "some writable dirs missing"
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)